source/material_model/grain_size.cc

```diff
@@ -270,9 +319,21 @@ namespace aspect
 
           // grain size growth due to Ostwald ripening
           const double m = grain_growth_exponent[phase_index];
-          const double grain_size_growth_rate = grain_growth_rate_constant[phase_index] / (m * std::pow(grain_size,m-1))
-                                                * std::exp(- (grain_growth_activation_energy[phase_index] + pressure * grain_growth_activation_volume[phase_index])
-                                                           / (constants::gas_constant * temperature));
+          double grain_size_growth_rate = 0.0;
+
+          // grain growth rate in the two-phase damage model
+          if (grain_size_evolution_formulation == Formulation::pinned_grain_damage)
+            grain_size_growth_rate = grain_growth_rate_constant[phase_index] * geometric_constant[phase_index] * phase_distribution_function (volume_fraction_phase_one)
+                                     / (m * std::pow(grain_size,m-1) * std::pow(roughness_to_grain_size_factor(volume_fraction_phase_one), m))
+                                     * std::exp(- (grain_growth_activation_energy[phase_index] + pressure * grain_growth_activation_volume[phase_index])
+                                                / (constants::gas_constant * temperature));
+          else
+            grain_size_growth_rate = grain_growth_rate_constant[phase_index] / (m * std::pow(grain_size,m-1))
+                                     * std::exp(- (grain_growth_activation_energy[phase_index] + pressure * grain_growth_activation_volume[phase_index])
+                                                / (constants::gas_constant * temperature));
```

Comment:
This looks correct to me, but I would implement it in fewer lines:

```suggestion
double grain_size_growth_rate = grain_growth_rate_constant[phase_index] / (m * std::pow(grain_size,m-1))
* std::exp(- (grain_growth_activation_energy[phase_index] + pressure * grain_growth_activation_volume[phase_index])
/ (constants::gas_constant * temperature));
                                                           
// in the two-phase damage model grain growth depends on the proportion of the two phases                         
if (grain_size_evolution_formulation == Formulation::pinned_grain_damage)
  grain_size_growth_rate *= geometric_constant[phase_index] * phase_distribution_function (volume_fraction_phase_one) / 
std::pow(roughness_to_grain_size_factor(volume_fraction_phase_one), m)
```
That also makes it easier to see what the differences are.